### PR TITLE
Product should not be required if ProductListing is disabled

### PR DESCRIPTION
**Pull request by riteshshrv**
[#126219995]

### CODE REVIEW

File: product.py [hunk 1]
```diff
@@ -169,20 +169,22 @@ class ProductSaleChannelListing(ModelSQL, ModelView):
...
-    # showing a manual channel is pretty much useless
+    # TODO: Only show channels where this ability is there.
     channel = fields.Many2One(
         'sale.channel', 'Sale Channel',
-        domain=[('source', '!=', 'manual')],
         required=True, select=True,
         ondelete='RESTRICT'
     )
     product = fields.Many2One(
-        'product.product', 'Product', required=True, select=True,
```
Comment (tarunbhardwaj) [inline product.py]: add migration to remove required from product

File: product.py [hunk 2]
```diff
@@ -169,20 +169,22 @@ class ProductSaleChannelListing(ModelSQL, ModelView):
     '''
     __name__ = 'product.product.channel_listing'
 
-    # TODO: Only show channels where this ability is there. For example
-    # showing a manual channel is pretty much useless
+    # TODO: Only show channels where this ability is there.
     channel = fields.Many2One(
         'sale.channel', 'Sale Channel',
-        domain=[('source', '!=', 'manual')],
```
Comment (tarunbhardwaj) [inline product.py]: No, leave it as it is.

File: product.py [hunk 3]
```diff
@@ -169,20 +169,22 @@ class ProductSaleChannelListing(ModelSQL, ModelView):
...
     )
     product = fields.Many2One(
-        'product.product', 'Product', required=True, select=True,
-        ondelete='CASCADE'
+        'product.product', 'Product', select=True,
+        states={'required': Eval('state') == 'active'},
+        ondelete='CASCADE', depends=['state']
     )
     product_identifier = fields.Char(
-        "Product Identifier", select=True, required=True
```
Comment (tarunbhardwaj) [inline product.py]: no need

File: product.py [hunk 4]
```diff
@@ -219,6 +221,17 @@ class ProductSaleChannelListing(ModelSQL, ModelView):
...
     )
 
+    @classmethod
+    def __register__(cls, module_name):
+        TableHandler = backend.get('TableHandler')
+        cursor = Transaction().cursor
+        table = TableHandler(cursor, cls, module_name)
+
+        # Remove not null constraint from 'product'
+        table.not_null_action('product', action='remove')
```
Comment (prakashpp) [inline product.py]: this should be after super

**Pull request merged**